aflr/api_request.py

```diff
@@ -4,52 +4,72 @@
 import os
 import requests
 from requests.exceptions import HTTPError
+from functools import lru_cache
 
 
 class APIRequest:
-    def __init__(self, api_key=None, api_base=None, **params):
-        self.api_key = api_key if api_key else aflr.api_key
-        self.api_base = api_base if api_base else aflr.api_base
-        if self.api_key == None or self.api_key == "your-key":
-            self.api_key = os.environ.get("aflr_key", None)
-        if self.api_key == None or len(self.api_key) < 30:
+    @lru_cache(maxsize=10, typed=False)
+    def _api_key_checker(api_key=None):
+        if api_key == None or api_key == "your-key":
+            api_key = os.environ.get("aflr_key", None)
+        if not isinstance(api_key, str):
+            raise TypeError("api_key must be of type string.")
+        if api_key == None or len(api_key) < 32:
             raise ValueError(
                 "Please specify a valid api_key or create one here:\nhttps://console.api.audio"
             )
-        if not isinstance(self.api_key, str):
-            raise TypeError("api_key must be of type string.")
 
-    def _build_header(self):
+    @classmethod
+    def _build_header(cls):
+        cls._api_key_checker(aflr.api_key)
         # add more headers for analytics in the future
-        return {"x-api-key": self.api_key}
+        return {"x-api-key": aflr.api_key}
 
-    def _post_request(self, json, url=None):
-        url = self.url if not url else url
-        headers = self._build_header()
+    @classmethod
+    def _post_request(cls, json, url=None):
+        url = url or f"{aflr.api_base}{cls.resource_path}"
+        headers = cls._build_header()
         r = requests.post(url=url, headers=headers, json=json)
-        self._expanded_raise_for_status(r)
+        cls._expanded_raise_for_status(r)
         return r.json()
 
-    def _get_request(self, url, path_param=None, request_params=None):
-        headers = self._build_header()  # DRY. To be changed.
+    @classmethod
+    def _get_request(cls, url=None, path_param=None, request_params=None):
+        url = url or f"{aflr.api_base}{cls.resource_path}"
+        if hasattr(cls, "file_url"):
+            url = f"{aflr.api_base}{cls.file_url}"
+
+        if hasattr(cls, "bg_url"):
```

Comment:
what will happen if a class have both file and bg urls and I wanna get the file url?

Reply:
removed this logic, no more checkoing for file_url. Will need to rethink Sound class probably